pkg/leader/leaderelector.go

```diff
@@ -54,13 +61,45 @@ type Elector struct {
 }
 
 func NewElector(kubeClient *kubernetes.Clientset, config Config, configMapName string, logger logrus.FieldLogger) *Elector {
+	logger = logger.WithField("configMap", configMapName)
 	return &Elector{log: logger, config: config, kube: kubeClient, configMapName: configMapName, isLeader: false}
 }
 
 func (l *Elector) IsLeader() bool {
 	return l.isLeader
 }
 
+// Wait for leader, run given function, drop leader and exit.
+func (l *Elector) RunWithLeader(ctx context.Context, run func() error) error {
```

Comment:
You implemented a mechanism to allow stopping this after a time, using context.
But since in the autoMigrationWithLeader you don't expose the context, no one can really use the context to stop the wait for the migration

Reply:
but u don't need to do it. It will be stopped by kubernetes if it will be stuck.

Reply:
that's why i didn't add context as var in autoMigrationWithLeader